stdlib/src/collections/optional.mojo

```diff
@@ -379,6 +379,21 @@ struct Optional[T: CollectionElement](
             return self._value[T]
         return default
 
+    fn __getitem__(owned self) raises -> T:
+        """Move the value out of the Optional consuming it. If the optional is
+        empty, raise ValueError.
+
+        Returns:
+            The value inside the Optional.
+
+        Raises:
+            ValueError on empty Optional.
+        """
+
+        if not self:
+            raise Error("ValueError: take() on empty Optional")
```

Comment:
To the stdlib team: Shall we have the same error system as Python has?
All ValueError, TypeError, etc? I am not sure if it is the greatest one

Reply:
For me, only error message, without ValueError is informative on its own